go/hyperkit.go

```diff
@@ -298,12 +298,7 @@ func (h *HyperKit) execute(cmdline string) error {
 
 	// Run
 	h.buildArgs(cmdline)
-	err = h.execHyperKit()
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return h.execHyperKit()
```

Comment:
I prefer the style in which all the actions+checks are written in the same manner:
```
if !something {
    return error
}
[... more...]
if !something_else {
    return error
}
return success
```
i.e. to not special case the final one as you are doing here (and in the other similar places). It makes the flow more consistent and reduces churn if you need to add a new thing after the final one.

Changing to the `if err := foo ; err != nil` style for these would be fine though.

Reply:
I concur on this, and on many regards I dislike the coding style in Go.

However, there is one good thing in Go, and it is that they tend to enforce one way to write things, enforced by tools such as golint etc.  And in that case the style that you and I prefer is not the standard one.

So for instance with
```go
	if err := h.execHyperKit(); err != nil {
		return err
	}

	return nil
```
I get
```
$ golint
hyperkit.go:317:2: redundant if ...; err != nil check, just return error instead.
```
So I would still claim this commit is in the right direction.

Reply:
I disagree with `golint` on this matter, normally I just have to stomach it but since I'm a maintainer on this project I don't have to in this case ;-). If the other maintainers disagree I'm happy to be overruled by them.

NB also this paragraph from https://github.com/golang/lint:
> The suggestions made by golint are exactly that: suggestions. Golint is not perfect, and has both false positives and false negatives. Do not treat its output as a gold standard. We will not be adding pragmas or other knobs to suppress specific warnings, so do not expect or require code to be completely "lint-free". In short, this tool is not, and will never be, trustworthy enough for its suggestions to be enforced automatically, for example as part of a build process. Golint makes suggestions for many of the mechanically checkable items listed in Effective Go and the CodeReviewComments wiki page.

In this particular case I believe the (usually good) recommendation to avoid conditionals which don't actually impact program flow does not always apply at the end of a function.

Reply:
I'm fine either way

Reply:
@ijc Yes, sure, golint guys _had_ to write that.  It's like warnings in C/C++, they're just that, warnings.  However all the Go projects I've seen so far (but I'm new to this language, granted) pay attention to go fmt, go vet, golint.

I'll remove this commit entirely.